### PR TITLE
Add option for HTTP basic auth

### PR DESCRIPTION
**Pull request by martin-helmich**
This PR add the option to configure HTTP basic auth (for cases when the PowerDNS API is hidden behind some non-standard authentication gateway).

It can be used with the `pdns.WithBasicAuthentication("username", "password")` client option.

### CODE REVIEW

**Comment by martin-helmich** [comment]
DO NOT MERGE (yet); rebase to current master once #15 is merged.

**Review by hensur — APPROVED** [review]

**Pull request merged**
